py/label_microservice/combined_model.py

```diff
@@ -0,0 +1,53 @@
+"""A combined model combines multiple models."""
+
+import logging
+
+from label_microservice import models
+
+class CombinedLabelModels(models.IssueLabelModel):
+  """Generate predictions with multiple models and then combine the results"""
+
+  def __init__(self, models=None):
+    # A list of models to generate predictions
+    self._models = models
+
+  def predict_issue_labels(self, title:str , text:str):
+    """Return a dictionary of label probabilities.
+
+    Args:
+      title: The title for the issue
+      text: The text for the issue
+
+    Return
+    ------
+    dict: Dictionary of label to probability of that label for the
+      the issue str -> float
+    """
+    if not self._models:
+      raise ValueError("Can't generate predictions; no models loaded")
+
+    predictions = {}
+    for i, m in enumerate(self._models):
+      logging.info(f"Generating predictions with model {i}")
+
+      latest = m.predict_issue_labels(title, text)
+
+      predictions = self._combine_predictions(predictions, latest)
+
+    return predictions
+
+
+  def _combine_predictions(self, left, right):
+    """Combine two sets of predictions by taking the max probability."""
+
+    results = {}
+    results.update(left)
+
+    for label, probability in right.items():
```

Comment:
Consider using python `groupby` example:

```py
from itertools import groupby

dict1 = {'bug': .9,
         'feature': .74,
         'comment': .12}

dict2 = {'jupyter': .2,
         'feature': .8,
         'docs': .45}

combined_data = sorted(list(dict1.items()) + list(dict2.items()))

combined_predictions = dict([max(v) for k,v in groupby(combined_data, key=lambda x: x[0])])

assert combined_predictions == {'bug': 0.9, 
                               'feature': 0.8,
                               'comment': 0.12, 
                               'docs': 0.45,  
                               'jupyter': 0.2}
```

Reply:
![image](https://user-images.githubusercontent.com/1483922/71432363-f7ed7500-268d-11ea-85b9-ae5e469eb66b.png)

Reply:
Thanks. So I initially tried it out; but then I realized it looks like we need to sort the items before calling groupby. So it seems like it would be less efficient because now we have to do a sort
as opposed to just going through both lists which is linear in time.